scalding-commons/src/main/scala/com/twitter/scalding/commons/source/LzoGenericScheme.scala

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.commons.source
+
+import scala.reflect.ClassTag
+
+import com.twitter.bijection._
+import com.twitter.chill.Externalizer
+import com.twitter.elephantbird.cascading2.scheme.LzoBinaryScheme
+import com.twitter.elephantbird.mapreduce.input.combine.DelegateCombineFileInputFormat
+import com.twitter.elephantbird.mapreduce.io.{ BinaryConverter, GenericWritable }
+import com.twitter.elephantbird.mapreduce.input.{ BinaryConverterProvider, MultiInputFormat }
+import com.twitter.elephantbird.mapreduce.output.LzoGenericBlockOutputFormat
+import com.twitter.elephantbird.mapred.output.DeprecatedOutputFormatWrapper
+
+import org.apache.hadoop.mapred.{ JobConf, OutputCollector, RecordReader }
+import org.apache.hadoop.conf.Configuration
+
+import cascading.tap.Tap
+import cascading.flow.FlowProcess
+
+/**
+ * Serializes BinaryConverters to JobConf.
+ */
+private[source] object ExternalizerSerializer {
+  def inj[T]: Injection[Externalizer[T], String] = {
+    import com.twitter.bijection.Inversion.attemptWhen
+    import com.twitter.bijection.codec.Base64
+
+    implicit val baseInj = JavaSerializationInjection[Externalizer[T]]
+
+    implicit val unwrap: Injection[GZippedBase64String, String] =
+      // this does not catch cases where it's Base64 but not compressed
+      // but the decompression injection will, so it's safe to do this
+      new AbstractInjection[GZippedBase64String, String] {
+        override def apply(gzbs: GZippedBase64String) = gzbs.str
+        override def invert(str: String) = attemptWhen(str)(Base64.isBase64)(GZippedBase64String(_))
+      }
+
+    Injection.connect[Externalizer[T], Array[Byte], GZippedBase64String, String]
+  }
+}
+
+private[source] object ConfigBinaryConverterProvider {
+  val ProviderConfKey = "com.twitter.scalding.lzo.converter.provider"
+}
+
+/**
+ * Provides BinaryConverter serialized in JobConf.
+ */
+private[source] class ConfigBinaryConverterProvider[M] extends BinaryConverterProvider[M] {
+  import ConfigBinaryConverterProvider._
+
+  private[this] var cached: Option[(String, BinaryConverter[M])] = None
+
+  override def getConverter(conf: Configuration): BinaryConverter[M] = {
+    val data = conf.get(ProviderConfKey)
+    require(data != null, s"$ProviderConfKey is not set in configuration")
+    cached match {
+      case Some((data, conv)) => conv
```

Comment:
I think we need `case Some((d, conv)) if d == data => conv` matching on data, I think, will just shadow the data on line 71.

I think we need this because without it, a different conf passed could give different data.